src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -287,4 +290,43 @@ private boolean hasCurrentPeriod(SmallFullLinks smallFullLinks) {
     private boolean hasPreviousPeriod(SmallFullLinks smallFullLinks) {
         return smallFullLinks.getPreviousPeriod() != null;
     }
+
+    /**
+     * Checks whether a conditional note exists when there is no balancesheet value for it
+     * If there is, the note is then deleted
+     *
+     * @param balanceSheet the populated balancesheet
+     * @param smallFullLinks the links used to determine if notes are present
+     * @param transactionId The id of the CHS transaction
+     * @param companyAccountsId The company accounts identifier
+     * @throws ServiceException if there's an error on submission
+     */
+    private void checkConditionalNotes(CompanyProfileApi companyProfileApi,
+                                       BalanceSheet balanceSheet, SmallFullLinks smallFullLinks,
+                                       String transactionId, String companyAccountsId) throws ServiceException {
+
+        if (isMultipleYearFiler(companyProfileApi)) {
+
+            if ((isDebtorsCurrentAmountNull(balanceSheet) || isDebtorsPreviousAmountNull(balanceSheet))
+                && smallFullLinks.getDebtorsNote() != null) {
+
+                debtorsService.deleteDebtors(transactionId, companyAccountsId);
+            }
+        } else {
+            if ((isDebtorsCurrentAmountNull(balanceSheet) && smallFullLinks.getDebtorsNote() != null)) {
+                debtorsService.deleteDebtors(transactionId, companyAccountsId);
+            }
+        }
+    }
+
+    private boolean isDebtorsCurrentAmountNull(BalanceSheet balanceSheet) {
+        return balanceSheet.getCurrentAssets() != null && balanceSheet.getCurrentAssets().getDebtors() != null
+            && (balanceSheet.getCurrentAssets().getDebtors().getCurrentAmount() == null || balanceSheet.getCurrentAssets().getDebtors().getCurrentAmount() == (0));
+    }
```

Comment:
It might read better with something like:
`Long currentDebtors = Optional.ofNullable(balanceSheet.getCurrentAssets().getDebtors().getCurrentAmount()).orElse(0L);
return currentDebtors.equals(0L))` 
same with the below method but its personal preference I will leave it up to you.

Reply:
Yeah I agree, will update.

Reply:
Updated